_studio/mfx_lib/encode_hw/hevc/linux/base/hevcehw_base_va_lin.cpp

```diff
@@ -461,4 +474,21 @@ uint32_t DDI_VA::ConvertRateControlMFX2VAAPI(mfxU16 rateControl, bool bSWBRC)
     return 0;
 }
 
+uint32_t DDI_VA::ConvertSliceStructureVAAPIToMFX(uint32_t structure)
+{
+    switch (structure)
+    {
+        case VA_ENC_SLICE_STRUCTURE_POWER_OF_TWO_ROWS:
+            return uint32_t(1);
+        case VA_ENC_SLICE_STRUCTURE_EQUAL_ROWS:
```

Comment:
I guess caps can return multiple supported structures in different bits. This function will not work with bit combinations.
No need to cast uint32_t(1)

Reply:
> I guess caps can return multiple supported structures in different bits. This function will not work with bit combinations.

I see no hint in the code that multiple supported structures are considered on msdk side.

Reply:
I mean in value from driver several bits can be set.

Reply:
I have same concern with with @lakulako,  driver caps actually can return multiple supported structure like this one. VA_ENC_SLICE_STRUCTURE_EQUAL_ROWS | VA_ENC_SLICE_STRUCTURE_MAX_SLICE_SIZE | VA_ENC_SLICE_STRUCTURE_EQUAL_MULTI_ROWS;

Reply:
@aleonenk , @DenWolf , could you please also review ? I vagally recall you came across the same in AVCe